### DISCUSSION TITLE
Werte Shelly EM3 werden nicht angezeigt.

### DISCUSSION THREAD

**Discussion by Hutzelbrot** [hoylabs/OpenDTU-OnBattery#338] [Q&A]
Ich komme einfach nicht weiter.....
Kann jemand helfen?
Mein erster Post bei github, Bitte um Nachsicht.

opendtu-onbattery-generic.bin 2023.07.24
ESP32-DOWD-V3 mit Display
(noch) keine Batterie
Homeassistant auf RasPi4 MQTT Add-On
Shelly Integration mit diversen Shellies u.a. Shelly EM3
Alle Werte der openDTU werden in HA über MQTT angezeigt
(1 device 20 entities)
Die Werte von Shelly EM3 werden weder in der Live-Ansicht (Leistung7Netz)
noch in HA angezeigt.
Was mach ich falsch? Kann der EM3 nicht gleichzeitig mit der Shelly Integration und MQTT umgehen? 

![MQTT Explorer](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/102532567/a0918629-88d9-4daf-9c4f-145338ac66ef)

**Comment by sylvio69** [comment]
Ist denn der Shelly in der DTU richtig konfiguriert (Einstellungen/Power Meter)? Abgesehen davon würde ich empfehlen dort HTTP/JSON zu konfigurieren. Mit einer halbwegs aktuellen Firmware auf dem Shelly steht dort der aktuelle Verbrauch über alle Phasen saldiert zur Verfügung (http://192.168.x.xxx/status). Sollte aber auch über Mqtt funktionieren wenn die Topic's stimmen. Also in etwa so: shellies/shellyem3-XXXXXXXXXX/emeter/0/power.

**Reply by Feldsalat** [reply]
Dazu hier mein Problem mit MQTT: #331 
Mit HTTP/JSON funktioniert es nun richtig.

**Comment by helgeerbe** [comment]
Topic 1: shellies/shellyem3-xxx/emeter/0/power
Topic: 2: shellies/shellyem3-xxx/emeter/1/power
Topic 3: shellies/shellyem3-xxx/emeter/2/power

Die OpenDTU saldiert auf:

Aber http/json wäre besser.
- http://shellyem3.fritz.box/status
- Basic auth
- user name admin
- dein password
- json path: total_power

**Reply by swingstate** [reply]
Kline mich hier kurz ein - @helgeerbe kannst Du erläutern warum json die bessere Option ist? Habe mit MQTT bisher keine Probleme, aber wenn es etwas zu optimieren gibt bin ich dabei.

**Reply by Hutzelbrot** [reply]
Ich bin zwar nicht helgeerbe, aber ich denke es ist besser weil damit OpenDTU-OnBattery auch ohne MQTT Broker funktioniert, da es die Daten direkt vom Shelly EM3 bezieht.

**Reply by harryheizer** [reply]
hallo - damit ist doch gemeint http://192.168.xxx.xx/status  richtig ? wobei das meine shelly3EM-Ip-Adresse ist - 
wenn ich das so eingebe - bekomme ich : [HttpPowerMeter] Couldn't find a value for phase 0 with Json query ""
und wenn ich hinter die IP noch .fritz.box/status eingebe - bekomme ich : HTTP Error connection refused
kann mir einer sagen wo mein Fehler ist ?
(Ps - ESP32 mit 7x HM800, nRF24L01+ SSD1306 Display + Shelly3EM) mit akt. FW von 23.03

**Reply by SW-Niko** [reply]
Was steht bei dir im JSON Pfad? Wenn ich das oben richtig lese dann solle da "total_power" eingetragen werden.

**Reply by Hutzelbrot** [reply]
Im JSON Pfad steht total_power ohne ""
Aber bei http://192.168.xxx.xx/status sehe ich die kompletten Parameter vom EM3; mindestens 50 Zeilen lang

**Reply by harryheizer** [reply]
Danke, habe den Fehler gefunden...
total_power musste nochmal eingetragen werden,  das hatte ich erst nicht gemacht,  weil es schon 'da stand'

**Reply by gitisgreat2023** [reply]
@Hutzelbrot 
Ich will auch über http/json die Daten abfragen.
Was muss ich einstellen damit http://shelly.fritz.box/status überhaupt was zeigt?
Jetzt zeigt er im browser "Not Found".
Ich habe den Shelly über LAN eingebunden.

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/138098278/0dc40459-8f89-4155-9f89-a99cc982aaf5)

**Reply by Hutzelbrot** [reply]
Wieso http://shelly.fritz.box/status? shelly ist keine IP.
 http://192.168.xxx.xx/status und 192.168.xxx.xxx ersetzen durch die IP von deinem 3EM.

**Reply by gitisgreat2023** [reply]
Habe es gelöst, beim 3EM Pro ist es jetzt anders:

http://shelly.fritz.box/rpc/Shelly.GetStatus und /em:0/total_act_power

Ich nutze die ns lookup vom Fritz!Box, ganz praktisch. Geht bei jedem Fritz!Box. Leider lässt sich .fritz.box nicht noch abkürzen zB auf fritz, hatte das beim support angefragt. Aber besser als IPs zu typen... :-)

Einzig was mir noch auffällt ist dass es mit 10 Sekunden refresht wird, ich meinte da geht auch 5 Sekunden. Aber weiss nicht mehr wie das geht... :-(

**Reply by spcqike** [reply]
@gitisgreat2023 Versuch mal nur „Shelly“. Zumindest bei mir muss ich die lokale Domänenadresse (bei dir „Fritz.box“) nicht explizit angeben. 

alternativ kannste auch einen eigenen DNS Server wie „pihole“ betreiben. Und dort Domains angeben und verwalten wie du möchtest. Https wird bei dir lokal zu Hause wohl eher kein Thema sein, von daher … kannst du auch „Shelly.home“ oder was auch immer in einem eigenen DNS Server erstellen.

**Reply by gitisgreat2023** [reply]
@spcqike ja klar, habe ich sogar schon mal so aufgesetzt und genutzt, aber mit pihole habe ich wieder eine Kompenente die dann ausfallen kan...
Nur shelly eingeben führt zu einer google suche...

Noch mal doppel gecheckt unter windows: cmd, nslook up shelly -> timeout, nslookup shelly.fritz.box liefert die IP.

Das ist auch der Grund dass ich seit heute einem Shelly 3EMpro habe... mein iobroker ist jetzt nur noch da für Datensammeln (der Hichi lass ich weiterlaufen, Plan B wenn mit dem Shelly was ist).

Bzgl update frequency: in den config json interval bei powermeter in zB 5 ändern
"powermeter": {
    "enabled": true,
    "verbose_logging": true,
    "interval": 10,

siehe [hier](https://github.com/helgeerbe/OpenDTU-OnBattery/issues/486)

**Comment by Hutzelbrot** [comment]
Vielen Dank für die schnelle  Hilfe. Mit MQTT war ich trotz der Tips nicht erfolgreich, aber http/json funktioniert.
